sdk/metric/pipeline.go

```diff
@@ -160,37 +160,49 @@ func (p *pipeline) produce(ctx context.Context) (metricdata.ResourceMetrics, err
 
 // pipelineRegistry manages creating pipelines, and aggregators.  Meters retrieve
 // new Aggregators from a pipelineRegistry.
-type pipelineRegistry[N int64 | float64] struct {
+type pipelineRegistry struct {
 	views     map[Reader][]view.View
 	pipelines map[Reader]*pipeline
 }
 
-func newPipelineRegistries(views map[Reader][]view.View) (*pipelineRegistry[int64], *pipelineRegistry[float64]) {
+func newPipelineRegistries(views map[Reader][]view.View) *pipelineRegistry {
 	pipelines := map[Reader]*pipeline{}
 	for rdr := range views {
 		pipe := &pipeline{}
 		rdr.register(pipe)
 		pipelines[rdr] = pipe
 	}
-	return &pipelineRegistry[int64]{
-			views:     views,
-			pipelines: pipelines,
-		}, &pipelineRegistry[float64]{
-			views:     views,
-			pipelines: pipelines,
-		}
+	return &pipelineRegistry{
+		views:     views,
+		pipelines: pipelines,
+	}
+}
+
+func (reg *pipelineRegistry) createInt64Aggregators(inst view.Instrument, instUnit unit.Unit) ([]internal.Aggregator[int64], error) {
+	return createAggregators[int64](reg, inst, instUnit)
+}
+
+func (reg *pipelineRegistry) createFloat64Aggregators(inst view.Instrument, instUnit unit.Unit) ([]internal.Aggregator[float64], error) {
+	return createAggregators[float64](reg, inst, instUnit)
+}
```

Comment:
These do not seem needed. Each call site can just as easily call `createAggregators[int64 | float64](reg, inst, instUnit)` directly.